diff --git a/node_modules/@shopify/react-native-skia/src/sksg/Container.native.ts b/node_modules/@shopify/react-native-skia/src/sksg/Container.native.ts
index f7e224f..86e9f46 100644
--- a/node_modules/@shopify/react-native-skia/src/sksg/Container.native.ts
+++ b/node_modules/@shopify/react-native-skia/src/sksg/Container.native.ts
@@ -10,6 +10,9 @@ import { visit } from "./Recorder/Visitor";
 import "../skia/NativeSetup";
 import "../views/api";
 
+// create local reference for `strictGlobal` option in Worklets
+const SkiaViewApi = globalThis.SkiaViewApi;
+
 const nativeDrawOnscreen = (
   nativeId: number,
   recorder: JsiRecorder,
diff --git a/node_modules/@shopify/react-native-skia/src/sksg/Container.web.ts b/node_modules/@shopify/react-native-skia/src/sksg/Container.web.ts
index ba0a82e..99be220 100644
--- a/node_modules/@shopify/react-native-skia/src/sksg/Container.web.ts
+++ b/node_modules/@shopify/react-native-skia/src/sksg/Container.web.ts
@@ -12,6 +12,9 @@ import { Container, StaticContainer } from "./StaticContainer";
 import "../skia/NativeSetup";
 import "../views/api";
 
+// create local reference for `strictGlobal` option in Worklets
+const SkiaViewApi = globalThis.SkiaViewApi;
+
 const drawOnscreen = (Skia: Skia, nativeId: number, recording: Recording) => {
   "worklet";
   const rec = Skia.PictureRecorder();
